### ISSUE TITLE
Lint the whole ANSIBLE_DIRECTORY instead of individual files

### ISSUE THREAD

**Issue by db0** [github/super-linter#2080]
**Is your feature request related to a problem? Please describe.**

The way the super-linter works, is that is sends to the linter each individual file to be linted. 
Unfortunately this conflicts with the way ansible-lint works which is aware of ansible structure and can determine the purpose of files and directories based on their name and parse everything in one go

This not only causes a ton of false-positives (as reported in #1671 but it also causes issues with role files etc) but it also makes ansible linting take a massive amount of time. I can lint my entire collection in a few seconds, compared to a couple of minutes through super-linter.

**Describe the solution you'd like**

I understand that the approach of this tool is to be able to lint only the changed files, but the ansible-linter approach needs to be taken properly into account.

A middle-of-the-road approach would be to provide an env variable to be used to mark that ansible should use the single-pass approach (e.g. ANSIBLE_SINGLE_PASS) which, when true, will run the ansible-lint just once against the whole folder.

**Comment by ferrarimarco** [comment]
@admiralAwkbar after experimenting a bit with ansible-lint, I've found out that it has an [auto-detection mechanism](https://github.com/ansible-community/ansible-lint/blob/55844616b37daffdbc9edb59b2a9b95591db63e7/src/ansiblelint/config.py#L14).

I've had some success pointing it to a specific directory that contains "everything" about Ansible (could be our `ANSIBLE_DIRECTORY`). Here's an example:

```
INFO     Discovered files to lint using: git ls-files --cached --others --exclude-standard -z
INFO     Excluded removed files using: git ls-files --deleted -z
INFO     Executing syntax check on docker/ansible/etc/ansible/playbooks/main.yaml (0.63s)
DEBUG    Examining docker/ansible/etc/ansible/roles/test_role/tasks/main.yaml of type tasks
DEBUG    Examining docker/ansible/etc/ansible/roles/test_role/tasks/setup-Debian.yaml of type tasks
DEBUG    Examining docker/ansible/etc/ansible/roles/test_role/vars/main.yaml of type vars
DEBUG    Ignored exception from VariableNamingRule.<bound method VariableNamingRule.matchyaml of var-naming: All variables should be named using only lowercase and underscores>: 'NoneType' object has no attribute 'keys'
DEBUG    Examining docker/ansible/etc/ansible/requirements.yml of type requirements
DEBUG    Examining docker/ansible/etc/ansible/roles/test_role/handlers/main.yaml of type handlers
DEBUG    Examining docker/ansible/etc/ansible/inventory/hosts.yml of type inventory
DEBUG    Examining docker/ansible/etc/ansible/roles/test_role/vars/default.yaml of type vars
DEBUG    Ignored exception from VariableNamingRule.<bound method VariableNamingRule.matchyaml of var-naming: All variables should be named using only lowercase and underscores>: 'NoneType' object has no attribute 'keys'
DEBUG    Examining docker/ansible/etc/ansible/roles/test_role/tasks/include-variables.yaml of type tasks
DEBUG    Examining docker/ansible/etc/ansible/playbooks/main.yaml of type playbook
DEBUG    Examining docker/ansible/etc/ansible/roles/test_role/defaults/main.yaml of type vars
DEBUG    Ignored exception from VariableNamingRule.<bound method VariableNamingRule.matchyaml of var-naming: All variables should be named using only lowercase and underscores>: 'NoneType' object has no attribute 'keys'
DEBUG    Examining docker/ansible/etc/ansible/roles/test_role/vars/Debian.yaml of type vars
DEBUG    Ignored exception from VariableNamingRule.<bound method VariableNamingRule.matchyaml of var-naming: All variables should be named using only lowercase and underscores>: 'NoneType' object has no attribute 'keys'
DEBUG    Examining docker/ansible/etc/ansible of type role
DEBUG    Examining docker/ansible/etc/ansible/roles/test_role/meta/main.yaml of type meta
```

What if we make this a special case (when we build the file list) where we run ansible-lint against the whole `ANSIBLE_DIRECTORY`, instead of scanning for files inside that directory?

Note: if we decide to do this, we might need to introduce an optional `ANSIBLE_ROLES_PATH` variable as well because ansible-lint might need it when linting playbooks that reference those roles.

**Comment by ferrarimarco** [comment]
@admiralAwkbar If you think this approach is fine, I can start hacking something.

**Comment by admiralAwkbar** [comment]
@ferrarimarco I think its a great approach, let me know if you have any issues, sorry have been slammed with other work :(

**Comment by ferrarimarco** [comment]
No worries, I can have a look at this.
